packages/peregrine/lib/store/actions/cart/asyncActions.js

```diff
@@ -134,51 +134,77 @@ export const addItemToCart = (payload = {}) => {
     };
 };
 
+/**
+ * @param payload.cartItemId {Number} the id of the cart item we are updating
+ * @param payload.item {Object} the new configuration item if changes are selected.
+ * TODO: Test for update without any changes
+ * TODO: Test for simple update without changes
+ */
 export const updateItemInCart = (payload = {}) => {
-    const { cartItemId, fetchCartId, item } = payload;
+    const {
+        cartItemId,
+        fetchCartId,
+        item,
+        productType,
+        quantity,
+        updateItem
+    } = payload;
     const writingImageToCache = writeImageToCache(item);
 
     return async function thunk(dispatch, getState) {
         await writingImageToCache;
         dispatch(actions.updateItem.request(payload));
 
-        const { cart, user } = getState();
+        const {
+            cart: { cartId },
+            user
+        } = getState();
         const { isSignedIn } = user;
 
         try {
-            let cartEndpoint;
-
-            if (!isSignedIn) {
-                const { cartId } = cart;
-
-                if (!cartId) {
-                    const missingCartIdError = new Error(
-                        'Missing required information: cartId'
-                    );
-                    missingCartIdError.noCartId = true;
-                    throw missingCartIdError;
-                }
-
-                cartEndpoint = `/rest/V1/guest-carts/${cartId}/items/${cartItemId}`;
+            if (productType === 'simple') {
```

Comment:
This check is fragile. `productType` uses the REST type. When we switch `getCartDetails` to GQL this may break so we should be aware.